internal/controller/controllers/bmo_utils.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type BMOUtils struct {
+	client.Client
+	log            logrus.FieldLogger
+	kubeAPIEnabled bool
+}
+
+func NewBMOUtils(client client.Client, log logrus.FieldLogger, kubeAPIEnabled bool) *BMOUtils {
+	return &BMOUtils{client, log, kubeAPIEnabled}
+}
+
+// ConvergedFlowAvailable checks the baremetal operator version and returns true if it's equal or higher than the minimal version for converged flow
+func (r *BMOUtils) ConvergedFlowAvailable(minimalVersionForConvergedFlow string) bool {
+	key := types.NamespacedName{
+		Name: "baremetal",
+	}
+	clusterOperator := &v1.ClusterOperator{}
+	if err := r.Client.Get(context.TODO(), key, clusterOperator); err != nil {
+		r.log.Errorf("Error querying api for baremetal operator status: %s", err)
+		return false
+	}
+	if len(clusterOperator.Status.Versions) == 0 {
+		r.log.Infof("no version found for baremetal operator")
+		return false
+	}
+	version := clusterOperator.Status.Versions[0].Version
+	r.log.Infof("The baremetal operator version is %s, the minimal version for the converged flow is %s", version, minimalVersionForConvergedFlow)
+
+	available := semver.New(version).Compare(*semver.New(minimalVersionForConvergedFlow)) >= 0
+	if available {
+		r.log.Infof("Converged flow enabled")
+	}
+	return available
+}
+
+// TODO: replace this with public function in BMO https://github.com/openshift/cluster-baremetal-operator/pull/261
+func (r *BMOUtils) GetIronicServiceURL() (string, error) {
+	provisioningInfo, err := r.readProvisioningCR()
+	if err != nil {
+		r.log.WithError(err).Error("unable to get provisioning CR")
+		return "", err
+	}
+	ironicIP, err := r.getIronicIP(*provisioningInfo)
+	if err != nil || ironicIP == "" {
+		r.log.WithError(err).Error("unable to determine Ironic's IP")
+		return "", err
+	}
+	return getUrlFromIP(ironicIP), nil
+}
+
+func (r *BMOUtils) readProvisioningCR() (*metal3iov1alpha1.Provisioning, error) {
+	// Fetch the Provisioning instance
+	instance := &metal3iov1alpha1.Provisioning{}
+	namespacedName := types.NamespacedName{Name: metal3iov1alpha1.ProvisioningSingletonName, Namespace: ""}
+	if err := r.Client.Get(context.TODO(), namespacedName, instance); err != nil {
+		return nil, errors.Wrap(err, "unable to read Provisioning CR")
+	}
+	return instance, nil
+}
+
+func (r *BMOUtils) getIronicIP(info metal3iov1alpha1.Provisioning) (string, error) {
+	// this is how BMO get the IronicIP
+	config := info.Spec
+	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+		return config.ProvisioningIP, nil
+	}
+	return r.getPodHostIP(info.Namespace)
+}
+
+func getUrlFromIP(ipAddr string) string {
+	if strings.Contains(ipAddr, ":") {
```

Comment:
can we use 
https://github.com/openshift/assisted-service/blob/master/internal/network/family_utils.go#L8